File: Web.API/WebApp.API/WebApp.API/Models/Employee.cs
using System.ComponentModel.DataAnnotations;

namespace WebApp.API.Models
{
    public class Employee
    {

        [Key]
        public int EmployeeId { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public string ImageUrl { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string Address { get; set; }
        public DateTime HireDate { get; set; }

        public string? PublicId { get; set; } 

        public ICollection<EmployeeDetail> EmployeeDetails { get; set; }
    }
}

File: web.UI/src/app/interface/Employee.ts
export interface IEmployee {
  employeeId?: number; // Optional if not provided during creation
  name: string;
  email: string;
  phone: string;
  dateOfBirth: string; // Ensure this format is consistent with the backend
  address: string;
  hireDate: string; // Ensure this format is consistent with the backend

  imageUrl?: string; // To store the image URL returned from the backend
 
}
